### PR TITLE
Entity descriptions

### LINKED ISSUE #2022: Add a "description" field to API objects that don't have one
Would it be possible to use the POST method of the [snapshot endpoint](https://github.com/lxc/lxd/blob/master/doc/rest-api.md#post-4) to be able to add some additional meta-data to the snapshot dict?

I saw comments were possible on 1.0
http://man7.org/linux/man-pages/man1/lxc-snapshot.1.html

Is there another way to add comments or something in the works?

### PR DESCRIPTION
**Pull request by albertodonato**
Add a `description` field to containers, networks, storage pools and volumes.
It can be changed  through edit operations.

Closes #2022

### CODE REVIEW

**Comment by brauner** [comment]
@stgraber, this bumps the db schema with each commit. Should we maybe just bump the whole db once. I take it that none of this will be backported  so this should be fine.

**Comment by brauner** [comment]
@albertodonato, thanks!

**Comment by stgraber** [comment]
Looks almost ready to merge, just a few things:
 - Despite my earlier comment, I agree with @brauner that we should probably just do a single DB update for this. The schema change and DB update could be pulled into a separate commit applied before the others (making the codebase still functional at that point since it's not a non-null column).
 - Changes to shared/api/ like this needs an API extension comment above the added field.

I'm also not sure how I feel about adding a new column to the default "lxc list" output. We are already fighting limited horizontal screen space and this will be taking quite a bit of space even if no container has it set.

**Comment by albertodonato** [comment]
@stgraber added the patch as first commit (although IMO it doesn't make much difference). I also reverted the lxc list ouput change, the "d" column is still available but not in the default output.

**Comment by brauner** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins; test this please

**Comment by albertodonato** [comment]
jenkins: test this please

**Comment by albertodonato** [comment]
jenkins: test this please

File: test/suites/config.sh [hunk 1]
```diff
@@ -225,3 +225,34 @@ test_config_profiles() {
...
+    lxc init testimage foo -s "lxdtest-$(basename "${LXD_DIR}")"
+    lxc config show foo | sed 's/^description:.*/description: bar/' | lxc config edit foo
+    lxc config show foo | grep -q 'description: bar'
+    lxc delete foo
+}
+
+test_config_edit_container_snapshot_pool_config() {
+    # shellcheck disable=2034,2039,2155
+    local storage_pool="lxdtest-$(basename "${LXD_DIR}")"
+
```
Comment (brauner) [inline test/suites/config.sh]: I hope these are all spaces otherwise someone one will be very unhappy. ;)

**Pull request merged**